packages/valory/skills/abstract_round_abci/tests/test_common.py

```diff
@@ -51,3 +63,367 @@ def test_random_selection() -> None:
+            # depending on the test's parametrization
+            if not observation:
+                last_iteration(gen)
+                sleep_mocked.assert_called_once_with(irrelevant_config["sleep_time"])
+                self.randomness_behaviour.context.randomness_api.increment_retries.assert_called_once()
+                return
+
+            last_iteration(gen)
+            send_a2a_transaction_mocked.assert_called_once()
+            wait_until_round_end_mocked.assert_called_once()
+            set_done_mocked.assert_called_once()
+
+    def test_clean_up(self) -> None:
+        """Test `clean_up`."""
+        self.randomness_behaviour.clean_up()
+        self.randomness_behaviour.context.randomness_api.reset_retries.assert_called_once()
+
+    def teardown(self) -> None:
+        """Teardown run after each test method."""
+        self.randomness_behaviour.context.randomness_api.increment_retries.reset_mock()
+
+
+class TestSelectKeeperBehaviour:
+    """Tests for `SelectKeeperBehaviour`."""
+
+    select_keeper_behaviour: DummySelectKeeperBehaviour
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup the test class."""
+        cls.select_keeper_behaviour = DummySelectKeeperBehaviour(
+            name="test",
+            skill_context=MagicMock(
+                params=BaseParams(
+                    name="test",
+                    skill_context=MagicMock(),
+                    service_id="test_id",
+                    consensus=dict(max_participants=1),
+                    **irrelevant_config,
+                )
+            ),
+        )
+
+    @mock.patch.object(random, "shuffle", lambda do_not_shuffle: do_not_shuffle)
+    @pytest.mark.parametrize(
+        "participants, blacklisted_keepers, most_voted_keeper_address, expected_keeper",
+        (
+            (
+                frozenset((f"test_p{i}" for i in range(4))),
+                frozenset(),
+                "test_p0",
+                "test_p1",
+            ),
+            (
+                frozenset((f"test_p{i}" for i in range(4))),
+                frozenset(),
+                "test_p1",
+                "test_p2",
+            ),
+            (
+                frozenset((f"test_p{i}" for i in range(4))),
+                frozenset(),
+                "test_p2",
+                "test_p3",
+            ),
+            (
+                frozenset((f"test_p{i}" for i in range(4))),
+                frozenset(),
+                "test_p3",
+                "test_p0",
+            ),
+            (
+                frozenset((f"test_p{i}" for i in range(4))),
+                frozenset((f"test_p{i}" for i in range(1))),
+                "test_p1",
+                "test_p2",
+            ),
+            (
+                frozenset((f"test_p{i}" for i in range(4))),
+                frozenset((f"test_p{i}" for i in range(4))),
+                "",
+                "",
+            ),
+        ),
+    )
```

Comment:
`participants` is constant, doesn't need to be an argument then either

suggestion to introduce `TestCase` here
```
@dataclass
def TestCase:
    name: str
    most_voted: str
    expected: str
    blacklisted: Set[str]
    participants: Frozenset[str] = frozenset(map(str, range(4)))
```
and then test case can be named e.,g.: "most voted keeper among blacklisted"